packages/@aws-cdk/aws-events-targets/lib/sqs.ts

```diff
@@ -58,15 +61,18 @@ export class SqsQueue implements events.IRuleTarget {
       })
     );
 
+    let fifoParameters = {};
+    if (!!this.props.messageGroupId) {
```

Comment:
I don't think `if (!!truthyexpression)` adds anything over `if (truthyexpression)`. It converts the expression to `bool` but that's not required for `if`, and it doesn't change the semantics. For example, `!!''` still evaluates to `false`.

We sometimes use `!!` in short-circuiting evaluation in combination with the `&&` operator, if the result of a function needs to be bool.

```ts
function optionalArrayContainsHello(xs?: string[]): boolean {
  return !!xs && xs.indexOf(x => x === 'hello') > -1;
}
```

In this case, we couldn't write `return xs && xs.find(x => x === 'hello') > -1;` because the type of that expression would be `undefined | boolean` as opposed to `boolean`.

Other than that case, I don't think the `!!` operator is really necessary.

Reply:
Agreed. 

Honestly, it was a copy-paste from a few lines later in this file. I didn't quite take the effort to clean it up 😅

I'm fixing it now.